kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala

```diff
@@ -606,6 +606,21 @@ object KyuubiConf {
     .version("1.2.0")
     .fallbackConf(LEGACY_ENGINE_SHARE_LEVEL)
 
+  val ENGINE_POOL_SIZE_THRESHOLD: ConfigEntry[Int] = buildConf("engine.pool.size.threshold")
+    .doc("This parameter is introduced as a server-side parameter, " +
+      "and controls the upper limit of the engine pool.")
+    .version("1.4.0")
+    .intConf
+    .checkValue(s => s > 0 && s < 32, "Invalid Engine Pool Threshold Size, " +
+      "should be between 0 and 33.")
+    .createWithDefault(9)
+
+  val ENGINE_POOL_SIZE: ConfigEntry[Int] = buildConf("engine.pool.size")
+    .doc("Max number of engines that can be provided to clients.")
```

Comment:
we can also describe the relationship between the threshold and itself, and also how it works, e.g.  if size <= 1 balalah.. if balalalah

Reply:
ok